test/powershell/engine/Module/ModuleCmdlet.Tests.ps1

```diff
@@ -0,0 +1,518 @@
+            Set-Content -Force -Path $modData.Path -Value $newModuleContent
+
+            $module = Import-Module $modData.Path -PassThru -Force
+            $module.ExportedFunctions.Count | Should -Be 2
+            $module.ExportedFunctions.Keys  | Should -Contain "Test-FirstModuleFunction"
+            $module.ExportedFunctions.Keys  | Should -Contain "Test-SecondModuleFunction"
+        }
+
+        It "Re-imports modules from file with new members when -Force is used with -AsCustomObject" {
+            $modData = New-TestModule -Content $simpleModContent
+            $module = Import-Module $modData.Path -PassThru
+
+            $module.ExportedFunctions.Count | Should -Be 1
+            $module.ExportedFunctions.Keys  | Should -Contain "Test-FirstModuleFunction"
+            $module.ExportedFunctions.Keys  | Should -Not -Contain "Test-SecondModuleFunction"
+
+            $newModuleContent = (Get-Content $modData.Path | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }"
+            Set-Content -Force -Path $modData.Path -Value $newModuleContent
+
+            $module = Import-Module $modData.Path -AsCustomObject -Force
+            $module.'Test-SecondModuleFunction'() | Should -BeExactly "SECONDSTRING"
+        }
+    }
+
+    Context "Classes in modules" {
+        BeforeAll {
+            SetupModules
+        }
+
+        AfterAll {
+            TearDownModules
+        }
+
+        It "Uses updated class definitions when the module is reloaded" {
+            $content = @'
+$passedArgs = $Args
+class Root { $passedArgs = $passedArgs }
+function Get-PassedArgsRoot { [Root]::new().passedArgs }
+function Get-PassedArgsNoRoot { $passedArgs }
+'@
+
+            $modData = New-TestModule -Content $content
+
+            Import-Module $modData.Path -ArgumentList 'value1'
+            $rootVal = Get-PassedArgsRoot
+            $noRootVal = Get-PassedArgsNoRoot
+            $rootVal | Should -BeExactly $noRootVal
+
+            Import-Module $modData.Path -ArgumentList 'value2'
+            $rootVal = Get-PassedArgsRoot
+            $rootNoVal = Get-PassedArgsNoRoot
+            $rootVal | Should -BeExactly $rootNoVal
+        }
+
+        It "Uses updated class definitions in later imports rather than cached values" {
+            $modSrc1 = @'
+class MyObj
+{
+    [string]$Name
+    MyObj()
+    {
+        $this.Name = "X"
+    }
+}
+'@
+
+            $modSrc2 = @'
+class MyObj
+{
+    [string]$Name
+    MyObj()
+    {
+        $this.Name = "Y"
+    }
+}
+'@
+            $modData = New-TestModule -Content $modSrc1
+
+            $mod1 = Import-Module $modData.Path -PassThru
+
+            Set-Content -Path $modData.Path -Value $modSrc2
+            $mod2 = Import-Module $modData.Path -PassThru
+
+            $firstTypes = $mod1.GetExportedTypeDefinitions()
+            $secondTypes = $mod2.GetExportedTypeDefinitions()
+
+            $firstTypes.MyObj.Equals($secondTypes.Sub) | Should -BeFalse
```

Comment:
Where does the `Sub` come from?

Reply:
It does with `-Force` though

Reply:
Is that the behaviour we are looking for?